yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -10,108 +10,84 @@
-    )
+from ..converter.converter.converter import DummmyParser
+from ..converter.converter.converter import Runner as ConvRunner
 
-    input_dict['detect.dat'] = input_dict['detect.dat'].format(
-        cyl_nr=param_dict['cyl_nr'],
-        cyl_nz=param_dict['cyl_nz'],
-        mesh_nx=param_dict['mesh_nx'],
-        mesh_ny=param_dict['mesh_ny'],
-        mesh_nz=param_dict['mesh_nz']
-    )
 
+def run_shieldhit(param_dict, raw_input_dict):
+    """Shieldhit runner"""
     # create temporary directory
     with tempfile.TemporaryDirectory() as tmp_output_path:
 
-        for config_filename in input_dict:
-            abs_input_path = os.path.join(tmp_output_path, config_filename)
-            with open(abs_input_path, 'w') as temp_input_file:
-                temp_input_file.write(input_dict[config_filename])
+        # digest dictionary with project data (extracted from JSON file)
+        # and generate SHIELD-HIT12A input files
+        conv_runner = ConvRunner(parser=DummmyParser(),
+                                 input_data=raw_input_dict,
+                                 output_dir=tmp_output_path)
+
+        conv_runner.run_parser()
 
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
+
         isRunOk = runner_obj.run(settings=settings)
         if not isRunOk:
             return None
 
         elapsed = timeit.default_timer() - start_time
         print("MC simulation took {:.3f} seconds".format(elapsed))
 
-        estimator = runner_obj.get_data()
-
-        return estimator
-
-
-if __name__ == '__main__':
-    sys.exit(run_shieldhit(sys.argv[1:]))
+        estimators_dict = runner_obj.get_data()
+
+        return dummy_convert_output(estimators_dict)
+
+
+def dummy_convert_output(estimators_dict):
+    """Dummy function for converting simulation output to dictionary"""
+    if not estimators_dict:
+        return {"message": "No estimators"}
+
+    # result_dict is the dictionary object, which is later converted to json
+    # to provide readable api response for fronted
+
+    # result_dict contains the list of estimators
+    result_dict = {"estimators": []}
+    for estimator_name, estimator_obj in estimators_dict.items():
+
+        # est_dict contains list of pages
+        est_dict = {
+            "name" : estimator_name,
+            "pages": []}
+        for page in estimator_obj.pages:
+
+            # handling 1 dimension page
+            if page.dimension == 1:
```

Comment:
In case of higher number of dimensions, return number of dimensions.
Later we will extend that data with path to the output file (when simulation will be running on SLURM)